api/all/src/main/java/io/opentelemetry/api/common/Attributes.java

```diff
@@ -49,6 +49,14 @@
   /** Returns a read-only view of this {@link Attributes} as a {@link Map}. */
   Map<AttributeKey<?>, Object> asMap();
 
+  /**
+   * Returns a {@link Attributes} instance containing only key-value pairs that are unique to this
+   * instance compared to the input.
+   *
+   * @param other The {@link Attributes} to remove from this instance.
+   */
+  Attributes removeAll(Attributes other);
```

Comment:
I think we need a default implementation of this, so as not to possibly break API compatibility with existing implementations. That implementation does not need to be optimal from a performance perspective.

Reply:
Two points:

1. This does not break (binary) compatibility.  Previous implementations will link just fine at runtime, as long as it's not called.   We can even use this to catch link exceptions and provide a default behavior.  It's an important technique to use (in extreme situations) for bin-compat.
2. No reason not to provide a default here, done!